### PR TITLE
KOGITO-3849: reflect changes in kogito-runtimes

### PR DESCRIPTION
**Pull request by kostola**
* [Related Jira task](https://issues.redhat.com/browse/KOGITO-3849)
* [Related PR in kogito-runtimes](https://github.com/kiegroup/kogito-runtimes/pull/915)

### CODE REVIEW

File: jobs-service/pom.xml [hunk 1]
```diff
@@ -13,6 +13,10 @@
   <description>Jobs Service (Timers and Async Jobs)</description>
 
   <dependencies>
+    <dependency>
+      <groupId>org.kie.kogito</groupId>
+      <artifactId>kogito-events-api</artifactId>
```
Comment (danielezonca) [inline jobs-service/pom.xml:18]: Why this change?
If it is only related to a transitive usage and `optional`, please add a comment to explain this dep

Comment (kostola) [reply jobs-service/pom.xml:18]: This dependency is needed here because `jobs-service` uses `org.kie.kogito.event.AbstractDataEvent`, contained in `kogito-events-api`, as base class for its events.

**Review by MarianMacik — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]
